### PR TITLE
Add hypothesis testing for and string methods to model config

### PR DESCRIPTION
**Pull request by valentin-krasontovitsch**
**Issue**
Cf. #4178 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/model_config.py [hunk 1]
```diff
@@ -222,11 +222,32 @@ def data_root(self):
...
+                f"obs_config_file: {self.obs_config_file},\n"
+                f"ens_path: {self.getEnspath()},\n"
+                f"run_path_format: {self.getRunpathFormat()},\n"
+                f"job_name_format: {self.getJobnameFormat()},\n"
+                f"run_path: {self.getRunpathAsString()},\n"
+                f"max_internal_submit: {self.get_max_internal_submit()},\n"
+                f"gen_kw_export_name: {self.getGenKWExportName()},\n"
+                f"time_map: {self.get_time_map()},\n"
+            )
+            if self._address()
```
Comment (oysteoh) [inline src/ert/_c_wrappers/enkf/model_config.py:242]: `if self._address()` then what? Is something missing here?

Comment (valentin-krasontovitsch) [reply src/ert/_c_wrappers/enkf/model_config.py:242]: it's perhaps an awkwardly formatted ternary thingy - 

`A if B else C` but written as

```
(AA
AAAA
AA
AAA
AAA
A)
if B
else C
```

Comment (oysteoh) [reply src/ert/_c_wrappers/enkf/model_config.py:242]: ah.... i didn't notice at first :facepalm:

File: src/ert/_c_wrappers/enkf/res_config.py [hunk 2]
```diff
@@ -418,6 +422,13 @@ def _alloc_from_dict(self, config_dict):
             job = self.site_config.job_list.get_job_copy(
                 job_description[ConfigKeys.NAME]
             )
+            if job is None:
+                err_msg = (
+                    f"Could not find job `{job_description[ConfigKeys.NAME]}` "
+                    "in list of installed jobs: "
+                )
+                err_msg += f"{list(self.site_config.job_list)}"
+                raise ValueError(err_msg)
```
Comment (oysteoh) [inline src/ert/_c_wrappers/enkf/res_config.py:431]: these two new value errors - they will change the current behavior correct? I guess in a good way, but just wanted to raise a couple of questions. How will this be presented to our users? Should we handle the exception further up in the stack raised here?

File: src/ert/_c_wrappers/enkf/model_config.py (hunk 1 above)
Comment (valentin-krasontovitsch) [reply src/ert/_c_wrappers/enkf/model_config.py:242]: no worries, it is strangely formatted : )

File: src/ert/_c_wrappers/enkf/res_config.py (hunk 2 above)
Comment (oysteoh) [reply src/ert/_c_wrappers/enkf/res_config.py:431]: If this replace a `util_abort` this is great stuff and an improvement! :tada:

Comment (valentin-krasontovitsch) [reply src/ert/_c_wrappers/enkf/res_config.py:431]: thank you : )

File: tests/test_config_parsing/test_res_config.py [hunk 3]
```diff
@@ -91,3 +94,26 @@ def test_res_config_parses_date():
     expected_ens_path = f"{expected_storage}/ensemble"
     assert res_config.model_config.getEnspath() == expected_ens_path
     assert res_config.model_config.getRunpathAsString() == expected_run_path
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+@given(config_dicts())
+def test_res_config_throws_on_missing_forward_model_job(config_dict):
+    cwd = os.getcwd()
+    filename = config_dict[pytest.TEST_CONFIG_FILE_KEY]
```
Comment (oysteoh) [inline tests/test_config_parsing/test_res_config.py:103]: depending on the discussion later today i guess the lines like this is not valid anymore?

Comment (valentin-krasontovitsch) [reply tests/test_config_parsing/test_res_config.py:103]: sure! i can fixup after the meeting, with whatever we arrive at, or perhaps do it in a separate PR, if Eivind's idea would like to live in its own PR : )

Comment (valentin-krasontovitsch) [reply tests/test_config_parsing/test_res_config.py]: it's gone! not needed in these tests anyways, so whatever we do elsewhere, it's removed here, and that is the right thing to do here.

File: src/ert/_c_wrappers/enkf/model_config.py [hunk 4]
```diff
@@ -222,11 +222,32 @@ def data_root(self):
...
+            else ""
+        )
+
     def __ne__(self, other):
         return not self == other
 
     def __eq__(self, other):
-        if self.data_root() != other.data_root():
+        if os.path.normpath(self.data_root()) != os.path.normpath(other.data_root()):
```
Comment (oysteoh) [inline src/ert/_c_wrappers/enkf/model_config.py:250]: what is the reason behind this change? Was it not working correctly before?

**Review by oysteoh — APPROVED** [review]
Please remember to squash your commits!
